lib/hook.js

```diff
@@ -62,22 +62,18 @@ Hook.prototype.error = function(err) {
  */
 Hook.prototype.serialize = function serialize() {
   return {
-    $$isPending: this.isPending(),
+    $$isPending: !!this.isPending(),
     $$titlePath: this.titlePath(),
     ctx:
       this.ctx && this.ctx.currentTest
         ? {
-            currentTest: {
-              title: this.ctx.currentTest.title,
-              [MOCHA_ID_PROP_NAME]: this.ctx.currentTest.id
-            }
+            currentTest: this.ctx.currentTest.serialize()
           }
         : {},
-    parent: {
-      [MOCHA_ID_PROP_NAME]: this.parent.id
-    },
+    parent: this.parent.serialize(),
+    state: this.state,
     title: this.title,
     type: this.type,
-    [MOCHA_ID_PROP_NAME]: this.id
+    [MOCHA_ID_PROP_NAME]: this[MOCHA_ID_PROP_NAME]
```

Comment:
`this.id` is the getter for `[MOCHA_ID_PROP_NAME]`, so why this change here?

Reply:
Reverted.